source/uwp/Renderer/lib/AdaptiveCardElement.cpp

```diff
@@ -130,6 +134,40 @@ namespace AdaptiveNamespace
         return S_OK;
     }
 
+    IFACEMETHODIMP AdaptiveCardElementBase::MeetsRequirements(_In_ ABI::AdaptiveNamespace::IAdaptiveFeatureRegistration* featureRegistration,
+                                                              _Out_ boolean* value)
+    {
+        *value = true;
+        for (const auto& requirement : *m_requires)
+        {
+            HString requirementName;
+            RETURN_IF_FAILED(UTF8ToHString(requirement.first, requirementName.GetAddressOf()));
+
+            const auto& requirementVersion = requirement.second;
+
+            HString featureVersion;
+            RETURN_IF_FAILED(featureRegistration->Get(requirementName.Get(), featureVersion.GetAddressOf()));
+            if (!featureVersion.IsValid())
+            {
+                // host doesn't provide this requirement
+                *value = false;
+                return S_OK;
+            }
+            else
+            {
+                // host provides this requirement, but does it provide an acceptible version?
+                const SemanticVersion providesVersion{HStringToUTF8(featureVersion.Get())};
+                if (providesVersion < requirementVersion)
+                {
+                    // host's provided version is too low
+                    *value = false;
+                    return S_OK;
+                }
+            }
+        }
+        return S_OK;
```

Comment:
Hmm... we have this logic in two places since it's already in the shared model. What do you think about calling GetSharedModel() and then using its version of the logic?

Reply:
it's perhaps a bit unwieldy, if only because of how UWP is structured. that said, it's probably not any more unwieldy than duplicating logic :)



---
In reply to: [275449169](https://github.com/Microsoft/AdaptiveCards/pull/2684#discussion_r275449169) [](ancestors = 275449169)